src/snarky.d.ts

```diff
@@ -1209,6 +1220,17 @@ declare class Ledger {
   };
 }
 
+declare const Test: {
```

Comment:
Can we also use this to test the full signature logic — ie expose signPayment and signDelegstion and verifySignsture(s) and ensure we can verify signatures created with ocaml with js and vice versa

Reply:
We already test that we can recreate and verify OCaml signatures, however with a different architecture: Currently we have hard-coded signature test vectors for payments / stake delegations where the signatures where created with the OCaml code (this was part of an existing test, so I just took that).

Assuming that we never change how how payments and delegations are signed in OCaml (which, AFAIU, is the goal), hard coding the reference signatures is equivalent to creating them in OCaml on the fly (and it's arguably more light-weight). However, if they should change at one point, the existing test won't detect the newly introduced inconsistency.

Reply:
Ah okay! Yes we need to maintain backwards compatibility so those signatures being hard-coded is actually better I think!